Source/Blazorise/Base/BaseComponent.cs

```diff
@@ -86,6 +86,30 @@ protected void ExecuteAfterRender( Func<Task> action )
             executeAfterRendereQueue.Enqueue( action );
         }
 
+        /// <summary>
+        /// Executes given action after the rendering is done.
+        /// </summary>
+        /// <remarks>Don't await this on the UI thread, because that will cause a deadlock.</remarks>
+        protected async Task<T> ExecuteAfterRender<T>( Func<Task<T>> action )
```

Comment:
Can you explain why this implementation of ExecuteAfterRender belongs in the root component? Seems like a specific code that should be better somewhere in RTE project.

Reply:
Its a generic method not specific for RTE only. My preference would be to make it an extension method, but because `ExecuteAfterRender` is `protected` thats not possible. I'm fine with also keeping this in the RTE only and then you can always move it to the base component when you need it

Reply:
Calling existing `ExecuteAfterRender` didn't work?

Reply:
No because I need async return value. So that's why I work with a `TaskCompletionSource<TResult>`

Reply:
Ah, that's why. Since `ExecuteAfterRender` is `protected` then it's best to create a new `BaseRichTextEditComponent` where you will inherit from `BaseComponent` and then move your implementation of `ExecuteAfterRender`. That way we will leave the root class clean. Is that OK?

Reply:
Ok thats fine by me. We should convince Microsoft on protected extension methods :)

Reply:
It is already possible to define friendly assemblies for `internal` methods :)

```cs
[assembly: InternalsVisibleTo( "Blazorise.AntDesign" )]
[assembly: InternalsVisibleTo( "Blazorise.Bootstrap" )]
[assembly: InternalsVisibleTo( "Blazorise.Bulma" )]
[assembly: InternalsVisibleTo( "Blazorise.Frolic" )]
[assembly: InternalsVisibleTo( "Blazorise.Material" )]
```